### PR TITLE
checklist initial

### PR DESCRIPTION
**Pull request by DmytroSvirsa**

### CODE REVIEW

File: checklist.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Check Your Code Against the Following Points
+
+## Code Style
+
+Make sure that you provide message when exception rises.
```
Comment (Y-Havryliv) [inline checklist.md:5]: raises

File: checklist.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+## Code Efficiency
+
+### Make sure that your methods work with **O(1)** time complexity.
+
+The exceptions:
+- `__init__` and `clear` methods 
+- handling the collision in any method
+- resizing the dictionary
+
+Bad example:
```
Comment (Y-Havryliv) [inline checklist.md:16]: you forgot a good one

**Review by Abnormaltype — CHANGES_REQUESTED** [review]
add this point:
If you wanna check your `__delitem__` method use these tests:

```
@pytest.mark.timeout(5)
def test_deletion():
    items = [(f"Element {i}", i) for i in range(1000)]
    dictionary = Dictionary()
    for key, value in items:
        dictionary[key] = value
    for key, value in items:
        assert dictionary[key] == value
    assert len(dictionary) == len(items)
    for key, value in items:
        del dictionary[key]
    print(len(dictionary))
```

Also, `delitem` should return nothing.

File: checklist.md (hunk 2 above)
Comment (DmytroSvirsa) [reply checklist.md:16]: it is nof fair to give tham the solution, no?

File: checklist.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        
+    # in any method that add or delete item we change self.length
+
+    def __len__(self) -> int:
+        return self.length
+```
+
+**Note:** any loop have O(n) time complexity (list comprehension is also a loop)
+
+If you want check your `__delitem__` method
```
Comment (Abnormaltype) [inline checklist.md:47]: Also, delitem should return nothing.

File: checklist.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        self.length = 0
+        self.hash_table: list = [None] * 8
+        
+    # in any method that add or delete item we change self.length
+
+    def __len__(self) -> int:
+        return self.length
+```
+
+**Note:** any loop have O(n) time complexity (list comprehension is also a loop)
```
Comment (Abnormaltype) [inline checklist.md:45]: add here note about O(1) at best

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

File: checklist.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+```python
+class Dictionary:
+    def __init__(self) -> None:
+        self.hash_table: list = [None] * 8
+
+    def __len__(self) -> int:
+        return len([cell for cell in self.hash_table if cell])
+```
+
+Good example:
```
Comment (Y-Havryliv) [inline checklist.md:30]: good examples are always before bad ones in other theories

**Review by Y-Havryliv — APPROVED** [review]

**Review by Abnormaltype — APPROVED** [review]

**Pull request merged**